src/globus_cli/commands/collection/show.py

```diff
@@ -0,0 +1,87 @@
+import click
+
+from globus_cli.endpointish import Endpointish
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import collection_id_arg, command
+from globus_cli.principal_resolver import default_identity_id_resolver
+from globus_cli.services.gcs import connector_id_to_display_name, get_gcs_client
+from globus_cli.termio import FORMAT_TEXT_RECORD, formatted_print
+from globus_cli.utils import filter_fields, sorted_json_field
+
+STANDARD_FIELDS = [
+    ("Display Name", "display_name"),
+    ("Owner", default_identity_id_resolver.field),
+    ("ID", "id"),
+    ("Collection Type", "collection_type"),
+    ("Storage Gateway ID", "storage_gateway_id"),
+    ("Connector", lambda x: connector_id_to_display_name(x["connector_id"])),
+    ("Allow Guest Collections", "allow_guest_collections"),
+    ("Disable Anonymous Writes", "disable_anonymous_writes"),
+    ("High Assurance", "high_assurance"),
+    ("Authentication Timeout", "authentication_timeout_mins"),
+    ("Multi-factor Authentication", "require_mfa"),
+    ("Manager URL", "manager_url"),
+    ("HTTPS URL", "https_url"),
+    ("TLSFTP URL", "tlsftp_url"),
+    ("Force Encryption", "force_encryption"),
+    ("Public", "public"),
+    ("Organization", "organization"),
+    ("Department", "department"),
+    ("Keywords", "keywords"),
+    ("Description", "description"),
+    ("Contact E-mail", "contact_email"),
+    ("Contact Info", "contact_info"),
+    ("Collection Info Link", "info_link"),
+    ("Google Cloud Storage Project", "policies.project"),
+]
+
+PRIVATE_FIELDS = [
+    ("Root Path", "root_path"),
+    ("Default Directory", "default_directory"),
+    ("Sharing Path Restrictions", sorted_json_field("sharing_restrict_paths")),
+    ("Sharing Allowed Users", "sharing_users_allow"),
+    ("Sharing Denied Users", "sharing_users_deny"),
+    ("Sharing Allowed POSIX Groups", "policies.sharing_groups_allow"),
+    ("Sharing Denied POSIX Groups", "policies.sharing_groups_deny"),
```

Comment:
Like the Google Cloud Storage Project field above, I need to either remove these, or support nested field specifications in the output formatter.